keras/wrappers/scikit_learn.py

```diff
@@ -210,34 +190,35 @@ def fit(self, x, y, sample_weight=None, **kwargs):
 
     def predict(self, x, **kwargs):
         """Returns the class predictions for the given test data.
-
         # Arguments
             x: array-like, shape `(n_samples, n_features)`
                 Test samples where `n_samples` is the number of samples
                 and `n_features` is the number of features.
             **kwargs: dictionary arguments
                 Legal arguments are the arguments
                 of `Sequential.predict_classes`.
-
         # Returns
             preds: array-like, shape `(n_samples,)`
                 Class predictions.
         """
         kwargs = self.filter_sk_params(Sequential.predict_classes, kwargs)
-        classes = self.model.predict_classes(x, **kwargs)
+        
+        # check if the model is sequential or functional
+        if isinstance(self.model, Sequential):
+            classes = self.model.predict_classes(x, **kwargs)
+        else:
+            classes = np.round(self.model.predict(x, **kwargs)).astype(int)
```

Comment:
np.rint I think does this in a single function without astype

Reply:
Thanks for reviewing! But I didn't figure out how to use np.rint to do it, since the elements in classes should be "int" instead of float. And the numpy documentation (https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.rint.html) says the type of np.rint out is the same as input, which means it might not be able to convert a float array to an int array. Am I correct?

I modified old version code by mistake, so there were lots of modifications of space. I modified the latest keras version code after I realized that. The latest commit is for the latest version code. Sorry for that.